sdk/kas_client.go

```diff
@@ -113,6 +113,125 @@ func (k *KASClient) unwrap(keyAccess KeyAccess, policy string) ([]byte, error) {
 	return key, nil
 }
 
+func (k *KASClient) getNanoTDFRewrapRequest(header string, kasURL string, pubKey string) (*kas.RewrapRequest, error) {
+	kAccess := keyAccess{
+		Header:        header,
+		KeyAccessType: "remote",
+		URL:           kasURL,
+		Protocol:      "kas",
+	}
+
+	requestBody := requestBody{
+		Algorithm:       "ec:secp256r1",
+		KeyAccess:       kAccess,
+		ClientPublicKey: pubKey,
+	}
+
+	requestBodyJSON, err := json.Marshal(requestBody)
+	if err != nil {
+		return nil, fmt.Errorf("Error marshaling request body: %w", err)
+	}
+
+	tok, err := jwt.NewBuilder().
+		Claim("requestBody", string(requestBodyJSON)).
+		IssuedAt(time.Now()).
+		Expiration(time.Now().Add(secondsPerMinute * time.Second)).
+		Build()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create jwt: %w", err)
+	}
+
+	signedToken, err := k.accessTokenSource.MakeToken(func(key jwk.Key) ([]byte, error) {
+		signed, err := jwt.Sign(tok, jwt.WithKey(key.Algorithm(), key))
+		if err != nil {
+			return nil, fmt.Errorf("error signing DPoP token: %w", err)
+		}
+
+		return signed, nil
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign the token: %w", err)
+	}
+
+	rewrapRequest := kas.RewrapRequest{
+		SignedRequestToken: string(signedToken),
+	}
+	return &rewrapRequest, nil
+}
+
+func (k *KASClient) makeNanoTDFRewrapRequest(header string, kasURL string, pubKey string) (*kas.RewrapResponse, error) {
+	rewrapRequest, err := k.getNanoTDFRewrapRequest(header, kasURL, pubKey)
+	if err != nil {
+		return nil, err
+	}
+	grpcAddress, err := getGRPCAddress(kasURL)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := grpc.Dial(grpcAddress, k.dialOptions...)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to kas: %w", err)
+	}
+	defer conn.Close()
+
+	ctx := context.Background()
+	serviceClient := kas.NewAccessServiceClient(conn)
+
+	response, err := serviceClient.Rewrap(ctx, rewrapRequest)
+	if err != nil {
+		return nil, fmt.Errorf("error making rewrap request: %w", err)
+	}
+
+	return response, nil
+}
+
+func (k *KASClient) unwrapNanoTDF(header string, kasURL string) ([]byte, error) {
+	keypair, err := ocrypto.NewECKeyPair(ocrypto.ECCModeSecp256r1)
+	if err != nil {
+		return nil, fmt.Errorf("ocrypto.NewECKeyPair failed :%w", err)
+	}
+
+	publicKeyAsPem, err := keypair.PublicKeyInPemFormat()
+	if err != nil {
+		return nil, fmt.Errorf("ocrypto.NewECKeyPair.PublicKeyInPemFormat failed :%w", err)
+	}
+
+	privateKeyAsPem, err := keypair.PrivateKeyInPemFormat()
```

Comment:
Do we have to get a pem format here?  Seems like we could have some unnecessary operations of getting a keypair converting it to pem then right back to the original type. 

Seems like we are converting it back a few lines down.

https://github.com/opentdf/platform/blob/771abd6423d15b7dc30ba4742348f0195ea36037/lib/ocrypto/ec_key_pair.go#L207-L224

Reply:
That's creating a shared (diffie-hellman) key based on the KAS public key, and the nano private key.  It's not converting a key pair to PEM and back again.  

For the more general question of PEM, I created an issue to revisit the PEM-to-binary performance implications. #860  You're right it may be beneficial to reduce the number of key format conversions, but it's not critical path to land-and-expand nanotdf support, so I'd call it outside the scope of this PR.